kernel/kernel-api/src/main/java/io/delta/kernel/TransactionBuilder.java

```diff
@@ -113,6 +114,19 @@ TransactionBuilder withTransactionId(
    * @since 3.3.0
    */
   TransactionBuilder withTableProperties(Engine engine, Map<String, String> properties);
+  
+  /**
+   * Unset the provided table properties on the table. If a property does not exist this is a no-op.
+   * For now this is only supported for user-properties (in other words, does not support 'delta.'
+   * prefixed properties). An exception will be thrown upon calling
+   * {@link TransactionBuilder#build(Engine)} if the same key is both set and unset in the same
+   * transaction.
+   *
+   * @param propertyKeys the table property keys to unset (remove from the table properties)
+   * @return updated {@link TransactionBuilder} instance.
+   * @throws IllegalArgumentException if 'delta.' prefixed keys are provided
+   */
+  TransactionBuilder withUnsetTableProperties(Set<String> propertyKeys);
```

Comment:
Open to ideas on the exact method name. I kept going back and forth between a few ideas

Reply:
withTablePropertiesRemoved? does it align with the DomainMetadata>

Reply:
Since domain metadata is on Txn it doesn't have the "with" since not a builder it's just `removeDomainMetadata`

Reply:
I'm good with `withTablePropertiesRemoved` though